lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBOStorageLocationDAOImpl.java

```diff
@@ -135,6 +135,12 @@ public Long create(StorageLocationSetting dto) {
 		if (dbo.getEtag() == null) {
 			dbo.setEtag(UUID.randomUUID().toString());
 		}
+
+		// PFLM-5985
+		if (dbo.getUploadType() == null) {
+			dbo.setUploadType(UploadType.S3);
+		}
+
```

Comment:
Remove DAO and SQL changes, because S3 might not be the right default value, and we have data in Prod that has nulls.

Reply:
Had a chat with John, who came up with the idea of adding UploadType.NONE. This would make UploadType null-safe while simultaneously creating a sensible default.